sky/utils/controller_utils.py

```diff
@@ -606,10 +606,13 @@ def maybe_translate_local_file_mounts_and_sync_up(task: 'task_lib.Task',
     # Step 5: Add the file download into the file mounts, such as
     #  /original-dst: s3://spot-fm-file-only-bucket-name/file-0
     new_file_mounts = {}
-    storage = task.storage_mounts[file_mount_remote_tmp_dir]
-    store_type = list(storage.stores.keys())[0]
-    store_prefix = store_type.store_prefix()
-    bucket_url = store_prefix + file_bucket_name
+    if copy_mounts_with_file_in_src:
+        # file_mount_remote_tmp_dir will only exist when there are files in
+        # the src for copy mounts.
+        storage = task.storage_mounts[file_mount_remote_tmp_dir]
+        store_type = list(storage.stores.keys())[0]
+        store_prefix = store_type.store_prefix()
+        bucket_url = store_prefix + file_bucket_name
     for dst, src in copy_mounts_with_file_in_src.items():
         file_id = src_to_file_id[src]
         new_file_mounts[dst] = bucket_url + f'/file-{file_id}'
```

Comment:
nit - For readability, move the loop inside the if condition?
```suggestion
        bucket_url = store_prefix + file_bucket_name
        for dst, src in copy_mounts_with_file_in_src.items():
            file_id = src_to_file_id[src]
            new_file_mounts[dst] = bucket_url + f'/file-{file_id}'
```